.github/workflows/pants.yaml

```diff
@@ -18,17 +18,34 @@ jobs:
   build:
     name: Perform CI Checks
     needs: org-check
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        # The python version to run Pants itself on.
+        python-version: [3.9]
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/cache@v2
-      id: cache
+    - uses: actions/checkout@v3
+    - uses: pantsbuild/actions/init-pants@v2
+      # This action bootstraps pants and manages 2-3 GHA caches.
+      # See: github.com/pantsbuild/actions/tree/main/init-pants/
       with:
-        path: |
-          ~/.cache/pants/setup
-          ~/.cache/pants/lmdb_store
-          ~/.cache/pants/named_caches
-        key: ${{ runner.os }}-
+        pants-python-version: ${{ matrix.python-version }}
+        # cache0 makes it easy to bust the cache if needed
+        # just increase the integer to start with a fresh cache
+        gha-cache-key: cache0-py${{ matrix.python-version }}
+        # The Kotlin backend doesn't use named caches. See the example-python repo for an example
```

Comment:
Is this statement true?

Reply:
Probably not. While the Kotlin backend does not, the JVM support library that it uses does:

```
$ git grep append_only_caches src/python/pants/backend/kotlin | wc -l
       0
$ git grep append_only_caches src/python/pants/backend/kotlin src/python/pants/jvm | wc -l
       7
```

Reply:
Hmm, so what should we be mixing into the cache keys here? In Python it's the lockfiles.

Reply:
I think append_only_caches is distinct from named_caches? I've just realized I'm not sure any more

Reply:
`append_only_caches` is the actual name of what "named caches" are. They are named (append-only) caches.